Spec/Auth.swift

```diff
@@ -3725,47 +3725,37 @@ class Auth : QuickSpec {
-                        realtime.auth.authorize { tokenDetails, error in
-                            if let error = error, (error as NSError).code != URLError.cancelled.rawValue {
-                                fail(error.localizedDescription); partialDone(); return
-                            }
-                            expect(tokenDetails).toNot(beNil())
-                            if tokenDetailsFirst == nil {
-                                tokenDetailsFirst = tokenDetails
-                            }
-                            else {
-                                tokenDetailsLast = tokenDetails
-                            }
-                            partialDone()
-                        }
-                        realtime.auth.authorize { tokenDetails, error in
-                            if let error = error, (error as NSError).code != URLError.cancelled.rawValue {
-                                fail(error.localizedDescription); partialDone(); return
-                            }
-                            expect(tokenDetails).toNot(beNil())
-                            if tokenDetailsFirst == nil {
-                                tokenDetailsFirst = tokenDetails
+                        let callback: (ARTTokenDetails?, Error?) -> Void = { tokenDetails, error in
+                            if let error = error {
+                                if (error as NSError).code == URLError.cancelled.rawValue {
+                                    expect(tokenDetails).to(beNil())
+                                }
+                                else {
+                                    fail(error.localizedDescription); partialDone(); return
+                                }
                             }
                             else {
+                                expect(tokenDetails).toNot(beNil())
                                 tokenDetailsLast = tokenDetails
                             }
                             partialDone()
                         }
+                        // One of them will be canceled by the connection:
+                        realtime.auth.authorize(callback)
+                        realtime.auth.authorize(callback)
                     }
 
-                    expect(tokenDetailsFirst?.token).toNot(equal(tokenDetailsLast?.token))
                     expect(realtime.auth.tokenDetails).to(beIdenticalTo(tokenDetailsLast))
                     expect(realtime.auth.tokenDetails?.token).to(equal(tokenDetailsLast?.token))
 
                     if let transport = realtime.internal.transport as? TestProxyTransport, let query = transport.lastUrl?.query {
                         expect(query).to(haveParam("accessToken", withValue: realtime.auth.tokenDetails?.token ?? ""))
                     }
                     else {
-                        XCTFail("MockTransport is not working")
+                        fail("MockTransport is not working")
```

Comment:
I'm interested as to why `XCTFail` was not suitable here. 🤔

Reply:
When we're using `Quick/Nimble` framework for tests we should use methods provided by them, the `fail(...)` method is a part of `Nimble` framework. The above change is correct IMO but there is a lot of `XCTFail` usages in this class that should be changed for code cohesion.